LoginClient/PipeClient.cs

```diff
@@ -0,0 +1,368 @@
+/// terminals around the ship, this should be accomplished via logging 
+/// into a central server to test user and password combinations 
+/// (you must have at least one administrator password setup)
+/// You must create a Server Client program it must use IPC to communicate.
+/// Your program must have a login that uses standard hashing techniques.
+/// 
+/// </summary>
+namespace LoginClient
+{
+    /// <summary>
+    /// Allows communication between the client and server pipes.
+    /// </summary>
+    class PipeClient
+    {
+        [DllImport("kernel32.dll", SetLastError = true)]
+        static extern SafeFileHandle CreateFile(
+           String pipeName,
+           uint dwDesiredAccess,
+           uint dwShareMode,
+           IntPtr lpSecurityAttributes,
+           uint dwCreationDisposition,
+           uint dwFlagsAndAttributes,
+           IntPtr hTemplate);
+
+        /// <summary>
+        /// Handles the messages recieved from a server pipe.
+        /// </summary>
+        /// <param name="message">The byte message recieved</param>
+        public delegate void MessageRecievedHandler(byte[] message);
+
+        /// <summary>
+        /// The event that is called whenever a message is recieved from the server pipe.
+        /// </summary>
+        public event MessageRecievedHandler MessageRecieved;
+
+        /// <summary>
+        /// Handles the server disconnected message.
+        /// </summary>
+        public delegate void ServerDisconnectedHandler();
+
+        /// <summary>
+        /// The Event that is called when the server pipe is severed.
+        /// </summary>
+        public event ServerDisconnectedHandler ServerDisconnected;
+
+        /// <summary>
+        /// Handles the client connected to the server message.
+        /// </summary>
+        public delegate void ConnectedToServerHandler();
+
+        /// <summary>
+        /// The event that is called when the client connects to the server.
+        /// </summary>
+        public event ConnectedToServerHandler ConnectedToServer;
+
+        const int BUFFER_SIZE = 4096;
+
+        FileStream stream;
+        SafeFileHandle handle;
+        Thread readThread;
+
+        /// <summary>
+        /// Is the client connected to a server pipe.
+        /// </summary>
+        private bool connected;
+
+        /// <summary>
+        /// The name of the pipe connected to the server.
+        /// </summary>
+        private string pipeName;
+
+        /// <summary>
+        /// Is the client logged in as the admin.
+        /// </summary>
+        private bool isLoggedIn;
+
+        /// <summary>
+        /// The methods for accessing the 
+        /// connected, pipename, and isLoggedOn properties.
+        /// </summary>
+        /// <returns></returns>
+        #region Accessors
+
+        public bool IsLoggedIn()
+        {
+            return isLoggedIn;
+        }
+
+        private void IsLoggedIn(bool isLoggedInOrNot)
```

Comment:
Potential bug.
"IsLoggedIn" sounds like a thing you call to check if its logged in.
But below it calls connect to server.
Make this Either: Connect, or IsLoggedIn. But not both.
Because if you call this with false, it wont actually disconnect.
So better to have a Connect, Disconnect, and IsLoggedIn that does NO changes.